python/paddle/fluid/contrib/slim/tests/QAT_mkldnn_int8_readme.md

```diff
@@ -1,58 +1,105 @@
+
+## 3. How to turn a QAT model into an INT8 MKL-DNN model?
+
+A QAT model can be transformed into an INT8 quantized model if it contains enough information about quantization scales for every quantized operator in the graph. The process of quantization is done by the `Qat2Int8MkldnnPass` pass which comprises several steps:
+
+### Gathering scales
+
+The information about the quantization scales is being collected from three types of operators:
+
+* `fake_quantize_moving_average_abs_max` - imitates INT8 quantization of FP32 tensors, but keeps quantized output values as floats; is used before quantized operator (e.g. `conv2d`) to gather scale information for the op's input.
+* `fake_dequantize_max_abs` - imitates dequantization of INT8 tensors back into floats; it is used after quantized operator, and contains scale used for the op's weights dequantization.
+* `fake_quantize_dequantize_moving_average_abs_max` - imitates immediate quantization and dequantization; it can be used after a quantized operator to get the scale value for the op's output.
 
 Notes:
 
-* MKL-DNN and MKL are required. The performance gain can only be obtained with AVX512 series CPU servers.
-* INT8 accuracy is best on CPU servers supporting AVX512 VNNI extension.
+1. As the next steps describe, quantization will be applied later to an optimized FP32 model. It means that quantization scales for inputs and outputs of each quantized operator have to be gathered for tensors which are inputs and outputs of already optimized or fused operators. For example, if a model contains the following sequence of tensors and operators in the graph
+   ```... → input1 → conv2d → output1 → batch_norm → output2 → relu → output3 → ...```
+   and we want to quantize the `conv2d` op, then after applying FP32 optimizations the sequence will become
+   ```... → input1 → conv2d → output3 → ...```
+   and the quantization scales have to be collected for the `input1` and `outpu3` tensors in the QAT model.
+2. Quantization of the following operators is supported: `conv2d`, `depthwise_conv2d`, `mul`, `fc`, `pool2d`, `reshape2`, `transpose2`, `concat`.
+
+### Removing fake operators
+
+All the `fake_quantize_*` and `fake_dequantize_*` operators are being removed from the graph.
+
+### Dequantizing weights
+
+Weights of `conv2d` and `mul` operators are assumed to be fake-quantized (quantized, but kept as floats) in QAT models. Here, the information about the scale from `fake_dequantize_max_abs` operators is used to fake-dequantize the weights back to the full float range of values. At this moment the model becomes an unoptimized clean FP32 inference model.
+
+### Optimizing FP32 graph
+
+A series of standard optimization passes are being applied to the FP32 graph. This gives us an optimized FP32 inference model and we can proceed with INT8 quantization.
+
```

Comment:
Is it useful to list the standard MKL-DNN optimization fusion passes? I don't know if it is common knowledge in PaddlePaddle? If it is, then don't have to add

Reply:
The list of passes is not obvious. However, I am not sure that this should be put in this document.